src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -139,6 +248,13 @@ private Task ConsumeChunkBytesAsync(ulong bytesConsumed)
 
                     await ConsumeChunkBytesAsync((ulong)bytesRead).ConfigureAwait(false);
 
+                    // If there's still space left in the destination buffer, try to fill it with additional chunk data
+                    // from the connection's buffer, in order to minimize the number of ReadAsync calls.
+                    if (bytesRead < destination.Length && _chunkBytesRemaining == 0)
```

Comment:
I'm a little confused, how can this happen?  Only when we needed to do another read to get the chunk's terminating \r\n?

Reply:
Ah, nevermind, I think I understand.